### PR TITLE
ap-5037: remove partner_means_assessment feature flag setting

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-5037)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by colinbruce — APPROVED** [review]
LGTM :+1:

File: config/locales/en/admin.yml [hunk 1]
```diff
@@ -72,7 +72,6 @@ en:
           allow_welsh_translation: Allow Welsh translation
           enable_ccms_submission: Allow CCMS submissions
           enable_evidence_upload: Enable the new evidence upload feature
-          partner_means_assessment: Enable Partner Means Assessment
```
Comment (naseberry) [inline config/locales/en/admin.yml:75]: Reminder to remove the relevant hint text, now on line 87

File: app/views/providers/start/index.html.erb [hunk 2]
```diff
@@ -14,9 +14,6 @@
 
       <p class="govuk-body"><%= t(".dont_use_service.list_title") %></p>
       <ul class="govuk-list govuk-list--bullet">
-        <% if !Setting.partner_means_assessment? %>
-          <li><%= t(".dont_use_service.partner_list_item") %></li>
```
Comment (naseberry) [inline app/views/providers/start/index.html.erb:18]: Can key value now be removed from https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/e8466643eaf1e973f4f9829d801ac0c804bb9bcb/config/locales/en/providers.yml#L1558

File: config/locales/en/admin.yml (hunk 1 above)
Comment (kmahern) [reply config/locales/en/admin.yml:75]: 👍 thanks!

File: app/views/providers/start/index.html.erb (hunk 2 above)
Comment (kmahern) [reply app/views/providers/start/index.html.erb:18]: 👍

**Review by naseberry — APPROVED** [review]

**Pull request merged**